mlflow/utils/databricks_utils.py

```diff
@@ -165,7 +168,11 @@ def is_in_databricks_job():
 
 
 def is_in_databricks_model_serving_environment():
```

Comment:
The naming isn't correct! Can you add comments that only use OAuth token if it is defined and environment is model serving because for customers not using langchain or not in RAG private preview oauth file will not be mounted

Reply:
Can we update the name of the function?

Reply:
my concern with changing the name is this will affect compatibility downstream in vector search

Reply:
right now, we have a working VS wheel so I don't want to break that

Reply:
@victorsun123 / @kevincherian-db - Can we release new version for VS? If this naming fix can be made it seems much cleaner and I expect this code to live beyond private preview

Reply:
Wasn't aware of the dependency. I'd recommend we leave a clear docstring in the function and leave the name as-is to avoid breaking things.

Reply:
Actually I would prefer wrapping this method now, old client (yanked verison) can keep using the `is_in_databricks_model_serving_environment()` and the new client can use `should_use_auto_oauth`. WDYT @victorsun123 / @dbczumar ?

Reply:
yeah actually that makes sense to me, and then in `get_databricks_host_creds` we will call `should_use_auto_oauth`. @dbczumar  any objections?